src/tool/cpp/cppwinrt/strings/base_implements.h

```diff
@@ -1223,6 +1223,13 @@ namespace winrt
         using implements_type = implements;
         using IInspectable = Windows::Foundation::IInspectable;
 
+#ifdef _DEBUG
+        implements() noexcept
+        {
+            WINRT_ASSERT(!is_stack_address(this));
```

Comment:
My one comment is that this check is expensive (three syscalls--one for each VirtualQuery).  I don't know whether it is _too_ expensive--that is for you to say.

Reply:
Yes, a little concerning but `winrt::implements` tends to be used for course grained allocations. Rarely thousands of small objects (delegates don't use winrt::implements). Plus, this only applies to debug builds. If there are complaints I can offer a macro to opt this out of debug builds.